src/screens/LiquidityPools/RemoveLiquidity.js

```diff
@@ -146,54 +147,53 @@ const AddLiquidityScreen = ({
     );
   }, [poolTokenAmount, obtainedTokenFieldsValid, poolTokenFieldValid]);
 
-  const onObtainedAssetValueChange = (newValue: string, tokenIndex: number) => {
-    const assetsAmounts = calculateProportionalRemoveLiquidityAssetValues(
-      pool,
-      parseFloat(newValue) || 0,
-      tokenIndex,
-      liquidityPoolsReducer,
-    );
+  const onPoolTokenAmountChange = (newValue: string) => {
+    const amounts = calculateProportionalAssetAmountsForRemoval(pool, newValue || '0', null, liquidityPoolsState);
+    setPoolTokenAmount(newValue);
+    setObtainedAssetsValues(amounts.pairTokens.map((amount) => formatAmount(amount)));
+  };
 
-    setPoolTokenAmount(formatAmount(assetsAmounts.pop()));
+  const onObtainedAssetValueChange = (tokenIndex: number, newValue: string, newPercent: number | void) => {
+    const amounts = calculateProportionalAssetAmountsForRemoval(pool, newValue || '0', tokenIndex, liquidityPoolsState);
 
-    const formattedAssetsValues = assetsAmounts
-      .map((amount, i) => i === tokenIndex ? newValue : formatAmount(amount));
-    setObtainedAssetsValues(formattedAssetsValues);
-  };
+    // Ensure we will withdraw everything if user want to remove max value.
+    if (newPercent === 100) {
+      onPoolTokenAmountChange(poolStats?.userLiquidityTokenBalance.toFixed() ?? '0');
+      return;
+    }
 
-  const onPoolTokenAmountChange = (newValue: string) => {
-    const assetsAmounts = calculateProportionalRemoveLiquidityAssetValues(
-      pool,
-      parseFloat(newValue) || 0,
-      pool.tokensProportions.length,
-      liquidityPoolsReducer,
-    );
-    setObtainedAssetsValues(assetsAmounts.map(formatAmount));
-    setPoolTokenAmount(newValue);
+    setPoolTokenAmount(formatAmount(amounts.poolToken));
+    setObtainedAssetsValues(amounts.pairTokens.map((amount, i) =>
+      i === tokenIndex ? newValue : formatAmount(amount),
+    ));
   };
 
   const renderTokenInput = (tokenIndex: number) => {
     const poolTokenSymbol = poolTokenData?.symbol;
     if (!poolTokenSymbol) return null;
-    const maxAmountBurned = poolStats?.userLiquidityTokenBalance || 0;
+    const maxAmountBurned = poolStats?.userLiquidityTokenBalance.toNumber() ?? 0;
```

Comment:
`toNumber`?

Reply:
This is a hard one, as it's hard to get 100% right. My aim is to cover 95% expected use cases.

Expected use cases:
* withdraw 100% amount 
* withdraw partial amount of given pair token value (e.g. withdraw 1 ETH of pool tokens) - while giving user rough estimate of allowed pair token withdrawal

Key findings:
* *maxAmountBurned* is feed into `ValueInput` for *pair* tokens
* `ValueInput` uses `number` math for checking allowed values (`Number(assetBalance) < Number(amount)`)
* I did not want to change `ValueInput` math as it might impact rest of the app
* for 100% withdrawals the pair token input will be truncated to 6 decimal places, which should be below any `number` math rounding down.

@jdnmr wdyt?